app/models/report.rb

```diff
@@ -1480,12 +1497,17 @@ def self.report_top_uploads(report)
     FROM uploads up
     JOIN users u
     ON u.id = up.user_id
-    WHERE up.created_at >= '#{report.start_date}' AND up.created_at <= '#{report.end_date}'
+    /*where*/
     ORDER BY up.filesize DESC
     LIMIT #{report.limit || 250}
     SQL
 
-    DB.query(sql).each do |row|
+    extension_filter = report.filter_values["file-extension"]
+    builder = DB.build(sql)
+    builder.where("up.created_at >= :start_date", start_date: report.start_date)
+    builder.where("up.created_at < :end_date", end_date: report.end_date)
+    builder.where("up.extension = :extension", extension: extension_filter) if extension_filter.present?
```

Comment:
do we need this `.present?`  we should build filter_values, to make sure `report.filter_values["xxx"]` returns nil of not present to avoid this everywhere

Reply:
> report.filter_values["xxx"] returns nil of not present to avoid this everywhere

So this is already the case, `filter_values` will return a `nil` if the key doesn't exist. Yet, the `.present?` here is a conditional for allowing the builder to specify the `where` statement itself.

If we don't have `.present?` then even if `report.filter_values["xxx"] returns nil`, we are still going to have a where condition that we don't need, which will result in wrong results.